cloud-controller-manager/do/loadbalancers.go

```diff
@@ -171,6 +172,18 @@ type loadBalancers struct {
 	lbActiveCheckTick int
 }
 
+type servicePatcher struct {
+	base *v1.Service
+}
+
+func newServicePatcher(base *v1.Service) servicePatcher {
+	return servicePatcher{base: base.DeepCopy()}
+}
+
+func (sp *servicePatcher) Patch(kclient kubernetes.Interface, updated *v1.Service) error {
+	return patchService(kclient, sp.base, updated)
```

Comment:
`updated` is always pointing at the same `base` Service object we received prior to deep-copying it, right?

If that's the case, then we could also a field

```
updated *v1.Service
``` 

to `servicePatcher` and set it to `base` (again, prior to deep-copying), thereby saving on passing through `updated` here.

Reply:
Nit: Similarly, we could instantiate `servicePatcher` with a `kubernetes.Interface`.